arroyo/processing/strategies/dead_letter_queue/dead_letter_queue.py

```diff
@@ -0,0 +1,54 @@
+import logging
+from typing import Optional
+
+from arroyo.processing.strategies.abstract import ProcessingStrategy as ProcessingStep
+from arroyo.processing.strategies.dead_letter_queue.policies.abstract import (
+    DeadLetterQueuePolicy,
+    InvalidMessage,
+)
+from arroyo.types import Message, TPayload
+from arroyo.utils.metrics import get_metrics
+
+logger = logging.getLogger(__name__)
+
+
+class DeadLetterQueue(ProcessingStep[TPayload]):
+    """
+    DLQ Processing Step.
+
+    Attempts to submit a given message to the next processing step,
+    handling an `InvalidMessage` according to the given Policy.
+    """
+
+    def __init__(
+        self,
+        next_step: ProcessingStep[TPayload],
+        policy: DeadLetterQueuePolicy[TPayload],
+    ) -> None:
+        self.__next_step = next_step
+        self.__policy = policy
+        self.__closed = False
+        self.__metrics = get_metrics()
+
+    def poll(self) -> None:
+        self.__next_step.poll()
```

Comment:
Thinking over this, we want to intercept the `InvalidMessage` exception here as well. 
Most consumers only do useful stuff per message in submit, but there are some that only accumulate messages in submit and do the work in `poll`